week1/matrix_done.py

```diff
@@ -0,0 +1,87 @@
+class Matrix:
+    def __init__(self, list):
+        self.rows= list
+    def print(self):
+        print('\n'.join(' '.join(map(str, sl)) for sl in self.rows))
+    def __eq__(self,other):
+        if self.rows==other.rows: return True
+        else: return False
+    def __add__(self, other):
+       m=Matrix([])
+       n=len(self.rows)
+       rows=[[0]*n for x in range(n)]
+       for i in range(0,n):
+          for j in range(0,n): rows[i][j]=self.rows[i][j]+other.rows[i][j]
+       m.rows=rows
+       return m
+    def __mul__(self, other):
+        ro=len(other.rows)
+        co=len(other.rows[0])
+        cs=len(self.rows[0])
+        rs=len(self.rows)
+        #print("row count="+str(r))
+        #print("colomn count=" + str(c))
+        m=Matrix.Zeros(ro,co)
+        if ro!=cs: raise ValueError("cannot multiply matrices with not matching columns and rows")
+        else:
+            for i in range(rs):
+                for j in range(co):
+                    sum=0
+                    for k in range(cs):
+                       sum=sum+self.rows[i][k]*other.rows[k][j]
+                       m.rows[i][j]=sum
+        return m
+    @staticmethod
```

Comment:
`@classmethod` is a much better fit here.

Reply:
Please read online about the difference between `staticmethod` vs `classmethod `. Methods `Zeroes` and `Identity` are semantically classmethods, so make them classmethods please.